handlers/download.go

```diff
@@ -69,68 +71,89 @@ func setStatusCode(ctx context.Context, w http.ResponseWriter, err error, logDat
 	http.Error(w, message, status)
 }
 
-// Do handle download dataset file requests. If the dataset is published and a public download link is available then
-// the request is redirected to the existing public link.
-// If the dataset is published but a public link does not exist then the requested file is streamed from the content
-// store and written to response body.
-// Authenticated requests will always allow access to the private, whether or not the version is published.
-func (d Download) Do(extension, serviceAuthToken, downloadServiceToken string) http.HandlerFunc {
+// DoImage handles download image file requests.
+func (d Download) DoImage(serviceAuthToken, downloadServiceToken string) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
-		ctx := req.Context()
 		params := getDownloadParameters(req, serviceAuthToken, downloadServiceToken)
-		logData := downloadParametersToLogData(params)
+		d.do(w, req, downloads.TypeImage, params, params.Ext, params.Variant, serviceAuthToken, downloadServiceToken)
+	}
+}
 
-		var err error
+// DoDatasetVersion handles dataset version file download requests.
+func (d Download) DoDatasetVersion(extension, serviceAuthToken, downloadServiceToken string) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		params := getDownloadParameters(req, serviceAuthToken, downloadServiceToken)
+		d.do(w, req, downloads.TypeDatasetVersion, params, extension, downloads.VariantDefault, serviceAuthToken, downloadServiceToken)
+	}
+}
 
-		datasetDownloads, err := d.DatasetDownloads.Get(ctx, params)
-		if err != nil {
-			setStatusCode(ctx, w, err, logData)
-			return
-		}
+// DoFilterOutput handles filter outpout download requests.
+func (d Download) DoFilterOutput(extension, serviceAuthToken, downloadServiceToken string) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		params := getDownloadParameters(req, serviceAuthToken, downloadServiceToken)
+		d.do(w, req, downloads.TypeFilterOutput, params, extension, downloads.VariantDefault, serviceAuthToken, downloadServiceToken)
+	}
+}
 
-		logData["published"] = datasetDownloads.IsPublished
-		log.Event(req.Context(), "attempting to get download", log.INFO, logData)
+// do handles download requests for any possible provided file type. If the object is published and a public download link is available then
+// the request is redirected to the existing public link.
+// If the object is published but a public link does not exist then the requested file is streamed from the content
+// store and written to response body.
+// Authenticated requests will always allow access to the private, whether or not the version is published.
+func (d Download) do(w http.ResponseWriter, req *http.Request, fileType downloads.FileType, params downloads.Parameters, extension, variant, serviceAuthToken, downloadServiceToken string) {
```

Comment:
`serviceAuthToken, downloadServiceToken` are not used in this function - should they be used?

Reply:
They are now part of the 'Parameters' struct - I will remove them from the parameter list, as they are no longer required there.
They are used in downloader.go like so:
```	image, err := d.ImageCli.GetImage(ctx, p.UserAuthToken, p.ServiceAuthToken, p.CollectionID, p.ImageID)
```